clar/summary.h

```diff
@@ -23,10 +23,11 @@ static int clar_summary_testsuite(struct clar_summary *summary,
     int idn, const char *name, time_t timestamp,
     int test_count, int fail_count, int error_count)
 {
-	struct tm *tm = localtime(&timestamp);
+	struct tm tm;
 	char iso_dt[20];
 
-	if (strftime(iso_dt, sizeof(iso_dt), "%Y-%m-%dT%H:%M:%S", tm) == 0)
+	localtime_r(&timestamp, &tm);
```

Comment:
Hm. The change makes sense, but I do wonder whether `localtime_r()` is even available on all relevant platforms. Our CI indicates that it's not:

```
D:/a/clar/clar/clar/summary.h:29:9: error: implicit declaration of function 'localtime_r'; did you mean 'localtime_s'? [-Werror=implicit-function-declaration]
   29 |         localtime_r(&timestamp, &tm);
      |         ^~~~~~~~~~~
      |         localtime_s
```

:(

Reply:
Good point, Visual C seems to offer [`localtime_s()`](https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/localtime-s-localtime32-s-localtime64-s?view=msvc-170), but not `localtime_r()`.

What do you think, should I test for the presence in CMake and then use a conditional `#if ... #else ... #endif` construct here? Or would you have a different preference?

Reply:
> What do you think, should I test for the presence in CMake and then use a conditional #if ... #else ... #endif construct here? Or would you have a different preference?

@dscho Ideally it'd work without relying on CMake. We do have an `#ifdef _WIN32` block in "clar.c". Can we maybe define a compatibility wrapper or something like that over there?

Reply:
It worked here: https://github.com/dscho/clar/actions/runs/14130165529

Reply:
@dscho Yup, this seems to work indeed. But let's move the declaration of `localtime_r` into "clar.c". "clar.h" is our public header, and there is no need to make this definition available to anyone outside of the clar itself. Something like this should work:

```patch
diff --git a/clar.c b/clar.c
index f345570..0bf6e41 100644
--- a/clar.c
+++ b/clar.c
@@ -48,6 +48,8 @@
 #	include <io.h>
 #	include <direct.h>
 
+#	define localtime_r(timer, buf) (localtime_s(buf, timer) == 0 ? buf : NULL)
+
 #	define _MAIN_CC __cdecl
 
 #	ifndef stat
```

Reply:
@dscho Yeah, it's used in "clar/summary.h". But that file gets included at the end of "clar.c", which is why it ultimately works. Anyway, this looks good to me now and passes CI. Giving @ethomson a couple days to chime in before I'm going to merge this. Thanks!